### PR TITLE
Add new ContinuousHikingAPI and ContinuousHikingCommandMessage

### PR DESCRIPTION
**Pull request by PotatoPeeler3000**
Updated the ContinuousWalkingAPI and message which are now deprecated. Created the ContinuousHikingAPI and ContinuousHikingCommandMessage.msg which will be what the process use's going forard. This contains the same stuff as the old message. Just a new name going forward

### CODE REVIEW

**Comment by PotatoPeeler3000** [comment]
Wanted to change the name to ContinuousHiking, but didn't want to remove the old one yet. This way people don't have to update the command until they are ready too.

File: ihmc-interfaces/src/main/messages/ihmc_interfaces/behavior_msgs/msg/ContinuousHikingCommandMessage.msg [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+bool use_monte_carlo_plan_as_reference
+
+# Walk with the Joystick Controller
+bool use_joystick_controller
+
+# Forward joystick value
+float64 forward_value
+
+# Backward walking from Joystick Controller
+bool walk_backwards
```
Comment (calvertdw) [inline ihmc-interfaces/src/main/messages/ihmc_interfaces/behavior_msgs/msg/ContinuousHikingCommandMessage.msg:36]: Why wouldn't backwards just be a negative forward_value?

Comment (PotatoPeeler3000) [reply ihmc-interfaces/src/main/messages/ihmc_interfaces/behavior_msgs/msg/ContinuousHikingCommandMessage.msg:36]: Don't know, I don't currently use the float forward_value. But I do use the boolean for walking backwards.

File: ihmc-interfaces/src/main/messages/ihmc_interfaces/behavior_msgs/msg/ContinuousHikingCommandMessage.msg [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+# Flag to square up to the goal when finishing walking. We may want the robot to end at an exact location.
+# Setting this to true tells the robot to step on the final goal positions.
+bool square_up_to_goal
+
+# Flag to enable/disable planning with A* planner. This is the default planner to use
+bool use_astar_footstep_planner
+
+# Flag to enable/disable planning with monte-carlo footstep planner
+bool use_monte_carlo_footstep_planner
```
Comment (ds58) [inline ihmc-interfaces/src/main/messages/ihmc_interfaces/behavior_msgs/msg/ContinuousHikingCommandMessage.msg:21]: Should the planner selection be a string or a short or something? Like, it wouldn't make sense to enable both astar and monte carlo.

Comment (PotatoPeeler3000) [reply ihmc-interfaces/src/main/messages/ihmc_interfaces/behavior_msgs/msg/ContinuousHikingCommandMessage.msg:21]: Maybe in the future that might make sense. I haven't put much thought into it. On one of my branches you can do this "hybrid" plan where if both of those are true. Then you do the Monte Carlo Planner and you use that plan as a reference to the A* Planner. That could probably be handled with a string or short,  I'm not trying to change that implementation to much at the moment. Much lower on the priority list then some other tasks.

Comment (PotatoPeeler3000) [reply ihmc-interfaces/src/main/messages/ihmc_interfaces/behavior_msgs/msg/ContinuousHikingCommandMessage.msg:21]: So for now I'd like to leave this, and if it causes problems going forward I can change it then.

**Review by ds58 — APPROVED** [review]

**Pull request merged**